cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -177,6 +177,7 @@ func removeIncompleteShards(dir string) {
 			if err := os.Remove(path); err != nil {
 				debug.Printf("failed to remove incomplete shard %s: %v", path, err)
 			} else {
+				os.Remove(path + ".meta")
```

Comment:
Same comment as above, can we factor out deletion to a little function that tries to delete both files?

Reply:
> Thinking about it, why should we keep the metadata inside the index file at all?

Agreed. We break some forward compatibility since older versions of zoekt rely on repoMetadata in the toc for in ReadMetadata. But that is fine since we mostly hack that in.

We chatted about this over zoom. To state what we decided we will remove repoMetadata from being in v17. Then every shard will _always_ have the "meta" file. This seems like a good design idea since it removes the state space of where metadata comes from. eg weird failure conditions like the meta file disappearing and falling back to the index time metadata.